### PR TITLE
[enocean] enable usage of Nodon roller-shutter repeater mode

### PR DESCRIPTION
**Pull request by SoCsOnFPGAsGuy**
For NODONs in Wall Modules SIN-2-2-00 and SIN-2-1-0x it is already possible to adjust the repeater mode. For NODONs roller-shutter (SIN-2-RS-01) this wasn't possibly with openhab so far - this series of commits enables the repeater mode setting in openhab also for NODONs roller-shutter modules, the same way it is done for the other two in Wall Modules.

Correctness also tested physically with some of the roller-shutter modules in my own house.

### CODE REVIEW

File: bundles/org.openhab.binding.enocean/src/main/java/org/openhab/binding/enocean/internal/eep/D2_05/D2_05_00_NodON.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    }
+
+    public D2_05_00_NodON(ERP1Message packet) {
+        super(packet);
+    }
+
+    @Override
+    protected void convertFromCommandImpl(String channelId, String channelTypeId, Command command,
+            Function<String, State> getCurrentStateFunc, Configuration config) {
+        if (channelId.equalsIgnoreCase(CHANNEL_REPEATERMODE)) {
```
Comment (lolodomo) [inline bundles/org.openhab.binding.enocean/src/main/java/org/openhab/binding/enocean/internal/eep/D2_05/D2_05_00_NodON.java:46]: `equals` would be ok

File: bundles/org.openhab.binding.enocean/src/main/java/org/openhab/binding/enocean/internal/eep/D2_05/D2_05_00_NodON.java [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+            if (command instanceof RefreshType) {
```
Comment (lolodomo) [inline bundles/org.openhab.binding.enocean/src/main/java/org/openhab/binding/enocean/internal/eep/D2_05/D2_05_00_NodON.java:47]: if (command == RefreshType.REFRESH) {

**Review by lolodomo — CHANGES_REQUESTED** [review]
Just 2 minor comments

**Review by lolodomo — APPROVED** [review]
LGTM

**Comment by lolodomo** [comment]
@fruggy83 ; is it ok for you?

File: bundles/org.openhab.binding.enocean/src/main/java/org/openhab/binding/enocean/internal/eep/D2_05/D2_05_00_NodON.java (hunk 1 above)
Comment (SoCsOnFPGAsGuy) [reply bundles/org.openhab.binding.enocean/src/main/java/org/openhab/binding/enocean/internal/eep/D2_05/D2_05_00_NodON.java]: fixed

File: bundles/org.openhab.binding.enocean/src/main/java/org/openhab/binding/enocean/internal/eep/D2_05/D2_05_00_NodON.java (hunk 2 above)
Comment (SoCsOnFPGAsGuy) [reply bundles/org.openhab.binding.enocean/src/main/java/org/openhab/binding/enocean/internal/eep/D2_05/D2_05_00_NodON.java]: fixed

File: bundles/org.openhab.binding.enocean/README.md [hunk 3]
```diff
@@ -224,9 +225,9 @@ If you change the SenderId of your thing, you have to pair again the thing with
 |                                 | suppressRepeating | Suppress repeating of msg   | true, false |
 | rollershutter                   | senderIdOffset    |                             | 1-127 |
 |                                 | enoceanId         | | |
-|                                 | sendingEEPId      |                             | A5_3F_7F_EltakoFSB, A5_3F_7F_EltakoFRM, A5_38_08_07, D2_05_00 |
+|                                 | sendingEEPId      |                             | A5_3F_7F_EltakoFSB, A5_3F_7F_EltakoFRM, A5_38_08_07, D2_05_00_NODON |
 |                                 | broadcastMessages |                             | true, false |
-|                                 | receivingEEPId¹   |                             | A5_3F_7F_EltakoFSB, A5_3F_7F_EltakoFRM, A5_11_03, D2_05_00 |
+|                                 | receivingEEPId¹   |                             | A5_3F_7F_EltakoFSB, A5_3F_7F_EltakoFRM, A5_11_03, D2_05_00_NODON |
```
Comment (fruggy83) [inline bundles/org.openhab.binding.enocean/README.md:230]: According to the rollershutter.xml the D2_05_00 EEP is still valid.

File: bundles/org.openhab.binding.enocean/src/main/java/org/openhab/binding/enocean/internal/eep/D2_05/D2_05_00_NodON.java [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+    }
+
+    public D2_05_00_NodON(ERP1Message packet) {
+        super(packet);
+    }
+
+    @Override
+    protected void convertFromCommandImpl(String channelId, String channelTypeId, Command command,
+            Function<String, State> getCurrentStateFunc, Configuration config) {
+        if (channelId.equals(CHANNEL_REPEATERMODE)) {
```
Comment (fruggy83) [inline bundles/org.openhab.binding.enocean/src/main/java/org/openhab/binding/enocean/internal/eep/D2_05/D2_05_00_NodON.java:46]: I know that channelId should currently never be NULL, however I would just switch the comparision.

**Review by fruggy83 — COMMENTED** [review]
@SoCsOnFPGAsGuy thanks a lot for your work and sorry for my late review. Maybe my comments are helpful for another PR in the future :)

@lolodomo Thanks a lot for your review and pushing this PR. Sorry, I just came back from a short trip today.

**Pull request merged**